libs/watt/src/lib/components/datetime-field/watt-datetime-field.component.ts

```diff
@@ -0,0 +1,204 @@
+
+const DA_FILLER = 'dd-mm-åååå, tt:mm';
+const EN_FILLER = 'dd-mm-yyyy, hh:mm';
+const DATETIME_FORMAT = 'DD-MM-YYYY, HH:mm';
+const PARTIAL_DATETIME_FORMAT = 'DD-MM-YYYY, ';
+const DANISH_TIME_ZONE_IDENTIFIER = 'Europe/Copenhagen';
+
+/* eslint-disable @angular-eslint/component-class-suffix */
+@Component({
+  standalone: true,
+  selector: 'watt-datetime-field',
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => WattDateTimeField),
+      multi: true,
+    },
+  ],
+  imports: [
+    ReactiveFormsModule,
+    MaskitoDirective,
+    MatCalendar,
+    WattButtonComponent,
+    WattFieldComponent,
+  ],
+  styles: [
+    `
+      watt-datetime-field {
+        display: block;
+        width: 100%;
+      }
+
+      .watt-datetime-field-picker {
+        position: fixed;
+        position-area: bottom span-right;
+        position-try-fallbacks: flip-block;
+        width: 296px;
+        height: 354px;
+        inset: unset;
+        margin: unset;
+        border: 0;
+      }
+    `,
+  ],
+  template: `
+    <watt-field
+      [label]="label()"
+      [control]="control"
+      [placeholder]="placeholder()"
+      [anchorName]="anchorName"
+    >
+      <input
+        #field
+        [formControl]="control"
+        [maskito]="mask"
+        (focus)="picker.showPopover()"
+        (blur)="handleBlur(picker, $event)"
+      />
+      <watt-button icon="date" variant="icon" (click)="field.focus()" />
+      <div
+        #picker
+        class="watt-elevation watt-datetime-field-picker"
+        popover="manual"
+        tabindex="0"
+        [style.position-anchor]="anchorName"
+      >
+        <mat-calendar
+          [startAt]="selected()"
+          [selected]="selected()"
+          (selectedChange)="handleSelectedChange(field, picker, $event)"
+        />
+      </div>
+    </watt-field>
+  `,
+})
+export class WattDateTimeField implements ControlValueAccessor {
+  private locale = inject(WattLocaleService);
+
+  // Popovers exists on an entirely different layer, meaning that for anchor positioning they
+  // look at the entire tree for the anchor name. This gives each field a unique anchor name.
+  private static instance = 0;
+  private instance = WattDateTimeField.instance++;
+  protected anchorName = `--watt-field-popover-anchor-${this.instance}`;
+
+  /** Converts date from outer FormControl to format of inner FormControl. */
+  protected modelToView = (value: Date | null, format = DATETIME_FORMAT) =>
+    value ? dayjs(value).tz(DANISH_TIME_ZONE_IDENTIFIER).format(format) : '';
+
+  /** Converts value of inner FormControl to type of outer FormControl. */
+  protected viewToModel = (value: string) => {
+    const date = dayjs(value, DATETIME_FORMAT, true);
+    if (!date.isValid()) return null;
+    return this.inclusive() ? date.endOf('m').toDate() : date.toDate();
```

Comment:
question: when `inclusive` is set, the initial control value should be end-of-day, right? `2024-11-20T00:00Z` will not be converted to ``2024-11-20T23:59:59.999Z``

Reply:
No, only "end-of-minute". I attempted to describe the behavior on the input [here](https://github.com/Energinet-DataHub/greenforce-frontend/pull/3761/files/97d3677ce46f9b2465eadcad6d5b8cc05ffbdddb#diff-c4c57765c0164d8b665398b0901bdfb7ae5ef6fc011d7959263ce16637e3bdf8R157), but maybe I need to make it clearer 🤔 I need this behavior so you can use two datetime fields as an alternative to a "datetime-range field", as the UX of such a field would be a nightmare to implement 😄

Reply:
Your example of `2024-11-20T00:00Z` should result in `2024-11-20T00:00:59.999Z` if using `[inclusive]=true`. I don't think it immediately converts the value and I'm not sure that it should, seems like the callers responsibility IMO 🙂